sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonReaderBuilder.java

```diff
@@ -427,6 +436,50 @@ private CarbonFileInputFormat prepareFileInputFormat(Job job, boolean enableBloc
     }
   }
 
+  private <T> void totalRowCountInSplits(Job job, List<InputSplit> splits,
+      List<Long> rowCountInSplit) throws IOException, InterruptedException {
+    long sum = 0;
+    boolean isIUDTable = false;
+    // Check if update or delete happened on the table.
+    if (!StringUtils.isEmpty(this.tablePath)) {
+      CarbonFile[] fileList = FileFactory.getCarbonFile(this.tablePath,
+          this.hadoopConf).listFiles();
+      for (CarbonFile file : fileList) {
```

Comment:
If you have too many files in table path, checking each file is delta file or not will take time, 
please manually check SDK update testcase and see if we write any metadata (like table status or update status) to know update has happened or not.

Reply:
Ok, Added an empty directory as a flag to check if update/delete happened for the non-transactional carbon table.